report.py

```diff
@@ -48,114 +49,74 @@ def run_report(config, blockers, server, header, test, save):
 			print('No OSP version could be found in job {}. Skipping...'.format(job_name))
 			continue
 
-		# get all relevant info from jenkins
-		try:
-			job_info = server.get_job_info(job_name)
-			job_url = job_info['url']
-			lcb_num = job_info['lastCompletedBuild']['number']
-			build_info = server.get_build_info(job_name, lcb_num)
-			build_actions = build_info['actions']
-			build_parameters = [action['parameters'] for action in build_actions if action.get('_class') == 'hudson.model.ParametersAction'][0]
-			run_mode = [action['text'] for action in build_actions if 'RUN_MODE: periodic' in action.get('text', '')]
-			gerrit_patch = [param['value'] for param in build_parameters if 'GERRIT_CHANGE_URL' in param.get('name', '')]
-
-			# find most recent build where the following is NOT true
-			# build has label "RUN MODE: periodic"
-			# build is associated with a gerrit patch (i.e. GERRIT_CHANGE_URL is present)
-			while run_mode != [] or gerrit_patch != []:
-				lcb_num = lcb_num - 1
-				build_info = server.get_build_info(job_name, lcb_num)
-				build_actions = build_info['actions']
-				build_parameters = [action['parameters'] for action in build_actions if action.get('_class') == 'hudson.model.ParametersAction'][0]
-				run_mode = [action['text'] for action in build_actions if 'RUN_MODE: periodic' in action.get('text', '')]
-				gerrit_patch = [param['value'] for param in build_parameters if 'GERRIT_CHANGE_URL' in param.get('name', '')]
-
-			lcb_url = build_info['url']
-			lcb_result = build_info['result']
-			compose = [action['text'][13:-4] for action in build_actions if 'core_puddle' in action.get('text', '')]
-
-			# No compose could be found; likely a failed job where the 'core_puddle' var was never calculated
-			if compose == []:
-				compose = "Could not find compose"
-			else:
-				compose = compose[0]
+		# get job info from jenkins API
+		jenkins_api_info = get_jenkins_job_info(server, job_name)
+
+		# if jeeves was unable to collect any good jenkins api info, skip job
```

Comment:
Wouldn't this leave the job name out of the report?

Reply:
It will, but if we can only get the job name and no other information there is not much benefit to including it in the report. It will however be reported in the logs, so in the event a job that we know should be included is missing we can refer to the logs to find out why it was not.